terraform/interpolate_test.go

```diff
@@ -136,6 +137,208 @@ func TestInterpolater_pathRoot(t *testing.T) {
+								"name_servers.3": "ns-601.awsdns-11.net",
+								"listeners.#":    "1",
+								"listeners.0":    "red",
+								"tags.#":         "1",
+								"tags.Name":      "reindeer",
+								"nothing.#":      "0",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	i := &Interpolater{
+		Module:    testModule(t, "interpolate-multi-vars"),
+		StateLock: lock,
+		State:     state,
+	}
+
+	scope := &InterpolationScope{
+		Path: rootModulePath,
+	}
+
+	name_servers := []string{
+		"ns-1334.awsdns-38.org",
+		"ns-1680.awsdns-18.co.uk",
+		"ns-498.awsdns-62.com",
+		"ns-601.awsdns-11.net",
+	}
+	expectedNameServers := strings.Join(name_servers, config.InterpSplitDelim)
+
+	// More than 1 element
+	testInterpolate(t, i, scope, "aws_route53_zone.yada.name_servers", ast.Variable{
+		Value: expectedNameServers,
+		Type:  ast.TypeString,
+	})
+
+	// Exactly 1 element
+	testInterpolate(t, i, scope, "aws_route53_zone.yada.listeners", ast.Variable{
+		Value: "red",
+		Type:  ast.TypeString,
+	})
+
+	// Zero elements
+	testInterpolate(t, i, scope, "aws_route53_zone.yada.nothing", ast.Variable{
+		Value: "",
+		Type:  ast.TypeString,
+	})
+
+	// Maps still need to work
+	testInterpolate(t, i, scope, "aws_route53_zone.yada.tags.Name", ast.Variable{
+		Value: "reindeer",
+		Type:  ast.TypeString,
+	})
+}
+
+func TestInterpolator_resourceMultiAttributesWithResourceCount(t *testing.T) {
+	i := getInterpolaterFixture(t)
+	scope := &InterpolationScope{
+		Path: rootModulePath,
+	}
+
+	name_servers := []string{
+		"ns-1334.awsdns-38.org",
+		"ns-1680.awsdns-18.co.uk",
+		"ns-498.awsdns-62.com",
+		"ns-601.awsdns-11.net",
+		"ns-000.awsdns-38.org",
+		"ns-444.awsdns-18.co.uk",
+		"ns-999.awsdns-62.com",
+		"ns-666.awsdns-11.net",
+	}
+
+	// More than 1 element
+	expectedNameServers := strings.Join(name_servers[0:4], config.InterpSplitDelim)
+	testInterpolate(t, i, scope, "aws_route53_zone.terra.0.name_servers", ast.Variable{
+		Value: expectedNameServers,
+		Type:  ast.TypeString,
+	})
+	// More than 1 element in both
+	expectedNameServers = strings.Join(name_servers[0:8], config.InterpSplitDelim)
+	testInterpolate(t, i, scope, "aws_route53_zone.terra.*.name_servers", ast.Variable{
+		Value: expectedNameServers,
+		Type:  ast.TypeString,
+	})
+
+	// Exactly 1 element
+	testInterpolate(t, i, scope, "aws_route53_zone.terra.0.listeners", ast.Variable{
+		Value: "red",
+		Type:  ast.TypeString,
+	})
+	// Exactly 1 element in both
+	testInterpolate(t, i, scope, "aws_route53_zone.terra.*.listeners", ast.Variable{
+		Value: strings.Join([]string{"red", "blue"}, config.InterpSplitDelim),
+		Type:  ast.TypeString,
+	})
+
+	// Zero elements
+	testInterpolate(t, i, scope, "aws_route53_zone.terra.0.nothing", ast.Variable{
+		Value: "",
+		Type:  ast.TypeString,
+	})
+	// Zero + zero elements
+	testInterpolate(t, i, scope, "aws_route53_zone.terra.*.nothing", ast.Variable{
+		Value: config.InterpSplitDelim, // Not sure if this is desired behaviour?
```

Comment:
I don't think it's worth rabbit holing too much on this, so I'm ultimately fine with it either way, but my vote would still be for a single list, since our "string-hack" lists don't allow for nesting right now.